src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs

```diff
@@ -89,7 +89,11 @@ public virtual Stream GetNuspec()
         {
             // This is the default implementation. It is overridden and optimized in
             // PackageArchiveReader and PackageFolderReader.
+            return GetStream(GetNuspecFile());
+        }
 
+        public virtual string GetNuspecFile()
+        {
             // Find all nuspecs in the root folder.
             var nuspecPaths = GetFiles()
                 .Where(entryPath => PackageHelper.IsManifest(entryPath))
```

Comment:
Noticed this clause from the old `GetNuspecFile` is missing:
```c#
.Select(f => f.TrimStart('/').Replace('/', Path.DirectorySeparatorChar))
```
Should it be moved to here as well?